server/src/internalClusterTest/java/org/elasticsearch/timeseries/support/TimeSeriesDimensionsLimitIT.java

```diff
@@ -0,0 +1,181 @@
+        final String dimensionFieldName = randomAlphaOfLength(randomIntBetween(513, 1024));
+        createTimeSeriesIndex(mapping -> {
+            mapping.startObject("routing_field").field("type", "keyword").field("time_series_dimension", true).endObject();
+            mapping.startObject(dimensionFieldName).field("type", "keyword").field("time_series_dimension", true).endObject();
+        },
+            mapping -> mapping.startObject("gauge").field("type", "integer").field("time_series_metric", "gauge").endObject(),
+            () -> List.of("routing_field"),
+            dimensionFieldLimit
+        );
+        final Exception ex = expectThrows(
+            DocumentParsingException.class,
+            () -> client().prepareIndex("test")
+                .setSource(
+                    "routing_field",
+                    randomAlphaOfLength(10),
+                    dimensionFieldName,
+                    randomAlphaOfLength(1024),
+                    "gauge",
+                    randomIntBetween(10, 20),
+                    "@timestamp",
+                    Instant.now().toEpochMilli()
+                )
+                .get()
+        );
+        assertThat(
+            ex.getCause().getMessage(),
+            equalTo(
+                "Dimension name must be less than [512] bytes but [" + dimensionFieldName + "] was [" + dimensionFieldName.length() + "]."
+            )
+        );
+    }
+
+    public void testDimensionFieldValueLimit() throws IOException {
+        int dimensionFieldLimit = 21;
+        createTimeSeriesIndex(
+            mapping -> mapping.startObject("field").field("type", "keyword").field("time_series_dimension", true).endObject(),
+            mapping -> mapping.startObject("gauge").field("type", "integer").field("time_series_metric", "gauge").endObject(),
+            () -> List.of("field"),
+            dimensionFieldLimit
+        );
+        long startTime = Instant.now().toEpochMilli();
+        client().prepareIndex("test")
+            .setSource("field", randomAlphaOfLength(1024), "gauge", randomIntBetween(10, 20), "@timestamp", startTime)
+            .get();
+        final Exception ex = expectThrows(
+            DocumentParsingException.class,
+            () -> client().prepareIndex("test")
+                .setSource("field", randomAlphaOfLength(1025), "gauge", randomIntBetween(10, 20), "@timestamp", startTime + 1)
+                .get()
+        );
+        assertThat(ex.getCause().getMessage(), equalTo("Dimension fields must be less than [1024] bytes but was [1025]."));
+    }
+
+    public void testTotalNumberOfDimensionFieldsLimit() {
+        int dimensionFieldLimit = 21;
+        final Exception ex = expectThrows(IllegalArgumentException.class, () -> createTimeSeriesIndex(mapping -> {
+            mapping.startObject("routing_field").field("type", "keyword").field("time_series_dimension", true).endObject();
+            for (int i = 0; i < dimensionFieldLimit; i++) {
```

Comment:
Maybe I'm reading this incorrectly. This creates a mapping with 21 dimension fields. But this should succeed? Only if there are 22 dimensions this should fail?

Reply:
There is the routing_field that is a dimension so it is actually 22. I need to include something in the routing path otherwise the test fails because there is no routing path.